src/components/Admin/AlterUsers.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { useSelector, useDispatch, shallowEqual } from 'react-redux'
+import { Table } from '../Table/Table';
+import { useState } from 'react';
+import SelectedUserModal from './SelectedUserModal';
+import { setSelectedAlterUser, deleteUnapprovedUser, setAlterUsers } from '../../store/reducers/adminQuery';
+
+const AlterUsers = ({users, approve}) => {
+
+    const dispatch = useDispatch()
+    const [showModal, setShowModal] = useState(false)
+    const { currentUser, selectedAlterUser } = useSelector(state => ({
+        selectedAlterUser: state.adminQuery.selectedAlterUser,
+        currentUser: state.user
+    }), shallowEqual)
+
+    const openModalAndAlterUser = (user)  => {
+        dispatch(setSelectedAlterUser({...user}))
+        setShowModal(true)
+    }
+
+    const deleteUser = (user) => {
+        if (window.confirm("Are you sure you want to delete this account?")) {
+            dispatch(disableButton())
+            dispatch(deleteUnapprovedUser(user))
+            dispatch(setAlterUsers(
+                users.filter(u => u.uid !== user.uid)
+            ))
+        }
+    }
+
+    const getRole = role => {
+        if (role.isAffiliate) return 'AFFILIATE'
+        if (role.isUser) return 'USER'
+        if (role.isTrader) return 'TRADER'
+        if (role.isSuperAdmin) return 'SUPER ADMIN'
+        if (role.isAdmin) return 'ADMIN'
+    }
+
+    const disableButton = user => (user.role.isSuperAdmin && selectedAlterUser.role && !selectedAlterUser.role.isSuperAdmin) || user.uid === currentUser.uid
```

Comment:
The first condition here is wrong. SuperAdmins can edit other SuperAdmins. Imagine if the client did have another superAdmin and later on wanted to demote him/her back to a normal Admin.

And I don't get why you're checking `selectedAlterUser.role`, `selectedAlterUser` will not be populated until a button is clicked...so checking it before a click of a button doesn't gave any data whatsoever.

so this should be : 

`const disableButton = user => (user.role.isSuperAdmin && !currentUser.role.isSuperAdmin) || user.uid === currentUser.uid`